projects/rocsparse/clients/benchmarks/CMakeLists.txt

```
@@ -1,252 +1,82 @@
-../testings/testing_check_matrix_ell.cpp
-../testings/testing_check_matrix_hyb.cpp
-../testings/testing_check_spmat.cpp
-../testings/testing_bsrpad_value.cpp
+target_include_directories(rocsparse-bench PRIVATE
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../include>
 )
 
-add_executable(rocsparse-bench ${ROCSPARSE_BENCHMARK_SOURCES} ${ROCSPARSE_CLIENTS_COMMON} ${ROCSPARSE_CLIENTS_TESTINGS})
-
-# Target compile options
-target_compile_options(rocsparse-bench PRIVATE -Wno-deprecated -Wno-unused-command-line-argument -Wall)
-
-# Internal common header
-target_include_directories(rocsparse-bench PRIVATE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../include>)
-if (BUILD_CLIENTS_ONLY)
-  target_include_directories(rocsparse-bench PRIVATE ${rocsparse_INCLUDE_DIRS}/rocsparse)
+# OpenMP RPATH handling for hip-clang
+if(NOT WIN32 AND OPENMP_FOUND)
+    if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+        if(TARGET OpenMP::OpenMP_CXX)
+            set_target_properties(rocsparse-bench PROPERTIES
+                BUILD_RPATH "${HIP_CLANG_ROOT}/lib"
+                INSTALL_RPATH "$ORIGIN/../llvm/lib"
+            )
+        elseif(TARGET OpenMP::omp)
+            if(DEFINED openmp_LIB_DIR)
+                set_target_properties(rocsparse-bench PROPERTIES
+                    BUILD_RPATH "${HIP_CLANG_ROOT}/${openmp_LIB_DIR}"
+                    INSTALL_RPATH "$ORIGIN/../llvm/${openmp_LIB_DIR}"
+                )
+            endif()
+        endif()
+    endif()
 endif()
 
-# Add OpenMP if available
-apply_omp_settings(rocsparse-bench)
+target_link_libraries(rocsparse-bench PRIVATE
+    roc::rocsparse
+    hip::host
+    hip::device
+)
 
-# Target link libraries
-target_link_libraries(rocsparse-bench PRIVATE roc::rocsparse hip::host hip::device ${ROCSPARSE_CLIENT_LIBS})
-if (rocsparseio_FOUND)
-  target_link_libraries(rocsparse-bench PRIVATE roc::rocsparseio)
+if(OPENMP_FOUND)
+    if(TARGET OpenMP::omp)
+        target_link_libraries(rocsparse-bench PRIVATE OpenMP::omp)
+    elseif(TARGET OpenMP::OpenMP_CXX)
+        target_link_libraries(rocsparse-bench PRIVATE OpenMP::OpenMP_CXX)
+    endif()
 endif()
 
-if (NOT WIN32)
-  if (BUILD_WITH_ROCTX)
+if(rocsparseio_FOUND)
+    target_link_libraries(rocsparse-bench PRIVATE roc::rocsparseio)
+endif()
 
+if(NOT WIN32 AND ROCSPARSE_ENABLE_MARKER)
     if(ROCTRACER_INCLUDE_DIR AND ROCTX_LIBRARY)
-      # Add the include directory
-      target_include_directories(rocsparse-bench PRIVATE ${ROCTRACER_INCLUDE_DIR})
-
-      # Link the rocTx lib
-      target_link_libraries(rocsparse-bench PRIVATE ${ROCTX_LIBRARY})
-      target_compile_definitions( rocsparse-bench PRIVATE ROCSPARSE_CLIENTS_WITH_ROCTX )
+        target_include_directories(rocsparse-bench PRIVATE ${ROCTRACER_INCLUDE_DIR})
+        target_link_libraries(rocsparse-bench PRIVATE ${ROCTX_LIBRARY})
+        target_compile_definitions(rocsparse-bench PRIVATE ROCSPARSE_CLIENTS_WITH_ROCTX)
     endif()
-  endif()
 endif()
```

```suggestion
target_link_libraries(rocsparse-bench PRIVATE ${rocTracer})
if(ROCSPARSE_ENABLE_MARKER)
    target_compile_definitions(rocsparse-bench PRIVATE ROCSPARSE_CLIENTS_WITH_ROCTX)
endif()
```

this works in other libraries because there is no header file dependency - so we may need to add that back